### DISCUSSION TITLE
Is Raspberry Pi not ideal for synth projects like this?

### DISCUSSION THREAD

**Discussion by briwil** [FundamentalFrequency/.github#63] [General]
I only ask this because I've been a little disappointed by the LMN's usability, I'm running on an 8gb RPi4 and still get quite a bit of stuttering/slow downs/hiccups.
Since I first Iearned about this project I've discovered the Dirtywave M8 tracker, which runs off of a Teensy 4.1; I have the headless version running off of a gameboy emulator case and it works phenomenaly, with the Teensy handling all of the audio processing and the raspberry pi just handling the display duties. And it kind of boggles my mind that the Teensy is able to handle so much flawlessly, which makes me question the Pi- is it because it has the full desktop up and running that makes it not ideal? I would imagine on a spec sheet that the pi would blow the Teensy out of the water, but maybe I'm wrong to assume that?

**Comment by stonepreston** [comment]
Can you be more specific about your issues? 

I notice in your previous post your case is fully enclosed. Its possible your Pi is getting hot and being throttled. The Pi 4 is very powerful. The tracktion engine can be a bit buggy at times, but if you are seeing consistent issues it could be other things. First thing I would do is try running outside of your case to see if thats causing issues.

**Reply by briwil** [reply]
I've since modified my case to remove all the sides, so I don't believe it's a heat issue. There is some serious audio lag when pressing notes unless I lower the buffer size in the settings, but that quickly turns into distorted audio, so it's hard to find a sweet spot in-between. I've also had freeze ups and stutters if I have, say, four tracks going simultaneously.

**Reply by stonepreston** [reply]
hmm. There could be some room for performance improvements (ie not loading the entire rpi UI, only load the system daemons/features required for the LMN UI to run).
